### ISSUE TITLE
VITON: An Image-based Virtual Try-on Network

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#162]
## TL; DR

- Represent person with pose heatmap, body shape and face/hair (identity).
- First generate coarse image with clothing mask, then refine with parametric TPS model-based deformation.

## Motivation

- Virtual try-on has great potential industrial usage, but prior works typically leverage 3D information, rendering the technique unapplicable in practice.

## Prior Works

- Conditional Analogy GAN
- Traditional computer graphics.

### Clothing-agnostic Person Representation

![image](https://user-images.githubusercontent.com/46308407/160330612-11dd8fa9-5845-4bd1-a7a8-c7066fa56ae4.png)

- Pose heatmap (18 key points) from a SOTA pose estimator.
- Rough body shape from a SOTA human parser (segmentation) without face and hair, which is separately modeled.
- Face and hair segment attained from human parser for ***identity preservation**.

### Two-stage Generation

![image](https://user-images.githubusercontent.com/46308407/160329870-bf196e4e-4472-4191-aecf-912c6fd310c6.png)

- (First stage) Conditioned on person representation and clothing image (front view), U-Net encoder-decoder generates **coarse result** as well as **clothing mask**.
- Coarse result is penalized by perceptual loss w.r.t. the reference image, and clothing mask is guided by L1 loss.

![image](https://user-images.githubusercontent.com/46308407/160330627-3df98358-6f3b-4d0f-b0e1-8466a6126e01.png)

- Warp the clothing image based on predicted clothing mask using parametric TPS. *(Thin Plate Spline)*
- (Second stage) Conditioned on warped clothing and coarse result from previous stage, learn blending coefficient $\alpha$ and *alpha-blend* the coarse result with warped clothing to reproduce details of the clothing with adequate deformations.

![image](https://user-images.githubusercontent.com/46308407/160330648-f135f210-bdd4-4b8f-a0f8-e639bf27f554.png)

### Training objective

- Coarse network
![image](https://user-images.githubusercontent.com/46308407/160330700-1421d957-3784-40d7-bbeb-87ebc2859663.png)

- Refinement network
![image](https://user-images.githubusercontent.com/46308407/160330718-0c6aeaf1-89d2-4709-917b-63e88e7a6f8e.png)

### Qualitative Baseline Comparison

![image](https://user-images.githubusercontent.com/46308407/160330750-5aadb080-d313-4e43-b2bd-6b561e8adee2.png)

### Qualitative Ablation

![image](https://user-images.githubusercontent.com/46308407/160330769-eaa6efc6-3286-4650-afba-e8dd3ba3330b.png)

### Quantitative Evaluation

![image](https://user-images.githubusercontent.com/46308407/160330787-2b191d41-d9cd-4f56-accb-7d3c92065617.png)

### In the Wild

![image](https://user-images.githubusercontent.com/46308407/160330812-c1aa5117-76a7-4c98-9444-3b98e2c4e319.png)

## Comments

- Method is a bit messy, but the result (at least from what we see) is quite promising.
- A lot of work must have been devoted (including the dataset contribution), so cheers to the authors.
